### PR TITLE
Fix for issue #1812 - SSH fingerprint verification

### PR DESCRIPTION
**Pull request by mnaiman**
add ability to check server fingerprint

### CODE REVIEW

**Comment by kenkendk** [comment]
Great fix.

I will see if I can update the UI to also grab the host key and insert it.

**Comment by mnaiman** [comment]
Web UI can work with these new switch, it was checked.
Only thing Im not sure is approach.
Current approach is, that if fingerprint is not specified it allows to connect to server.
Better approach could be to refuse to connect to server if not specified or to add new switch like --shh-allow-all-fingerprints.
This could be secure by design. I can alter commit to this if we agree.

**Comment by kenkendk** [comment]
Yes, that is how the other SSH clients are doing, so we should do the same.
Then the WebUI can just connect without a fingerprint and the error message will contain the fingerprint.

**Comment by mnaiman** [comment]
Yeah, error contains fingerprint right now, so it is easy change. Do you want me to do it or you are working on it?

**Comment by agrajaghh** [comment]
just to clairfy, you want to accept the (unknown) fingerprint on the first connection and refuse the connection if it changes, right? so TOFU (trust on first use)? Because `--shh-allow-all-fingerprints` sounds different/dangerous

**Comment by kenkendk** [comment]
@agrajaghh My suggestion was to not trust anything, but instead report the fingerprint in an error message. On the CLI the message would be something like `please add --ssh-fingerprint=xyz to trust this host key`, in the WebUI it would be something like `Would you like to trust the SSH Host key xyz ?`.

To get the TOFU style, we would need to store the fingerprint somewhere, either in some local storage folder, or in the local database.

**Comment by mnaiman** [comment]
Updated to not trust anything and reporting server fingerprint in error message.

File: Duplicati/Library/Backend/SSHv2/SSHv2Backend.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class SSHv2 : IStreamingBackend, IRenameEnabledBackend, IBackend, IDispos
     {
         public const string SSH_KEYFILE_OPTION = "ssh-keyfile";
         public const string SSH_KEYFILE_INLINE = "ssh-key";
+        public const string SSH_FINGERPRINT_OPTION = "ssh-fingerprint";
+        public const string SSH_FINGERPRINT_ACCEPT_ANY_OPTION = "shh-accept-any-fingerprints";
```
Comment (kenkendk) [inline Duplicati/Library/Backend/SSHv2/SSHv2Backend.cs]: Small typo: should be `ssh-accept-fingerprints` instead of `shh...`

**Pull request merged**